### PR TITLE
sea3d revision 4

### PR DESCRIPTION
**Pull request by sunag**
- legacy support
- fixes in animation (old system yet)
- fix mobile skinned
- skeleton animation time sync with keyframe animation
- fast int24 read

### CODE REVIEW

File: examples/js/ShaderTerrain.js [hunk 1]
```diff
@@ -84,8 +84,6 @@ THREE.ShaderTerrain = {
 			"varying vec3 vNormal;",
 			"varying vec2 vUv;",
 
-			"uniform vec3 ambientLightColor;",
```
Comment (mrdoob) [inline examples/js/ShaderTerrain.js]: Hmm, I can see `ambientLightColor` being references in the file...?

Comment (sunag) [inline examples/js/ShaderTerrain.js]: Not in sea3d, It was harnesses the PR to fix the example webgl_terrain_dynamic.html

`THREE.WebGLProgram: shader error:  0 gl.VALIDATE_STATUS false gl.getProgramInfoLog invalid shaders  ERROR: 0:158: 'ambientLightColor' : redefinition`

Comment (mrdoob) [inline examples/js/ShaderTerrain.js]: Ah, I see. It's because it's already defined in `lights_pars.glsl`...

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**